astropy/units/tests/test_quantity_non_ufuncs.py

```diff
@@ -1006,8 +1006,9 @@ def test_nanargmax(self):
     def test_nanmean(self):
         self.check(np.nanmean)
 
-    def test_nanmedian(self):
-        self.check(np.nanmedian)
+    @pytest.mark.parametrize("axis", [None, 0, ~0])
```

Comment:
Maybe just `-0` instead of `~0`? That's not an official input for axes! (even if formally equal to -1).